helpers/mongoDB-utils.js

```diff
@@ -35,17 +40,15 @@ export const getAllRecipes = async (client, skip, limit) => {
   }
 };
 
-export const fetchRecipeDataFromMongo = async (recipeName, collection) => {
+export const fetchRecipeDataFromMongo = async (recipeName,collection) => {
   try {
     // Establish a connection to MongoDB and select your database and collection
     const client = await DBConnection();
     const db = client.db("devdb");
     const collec = db.collection(collection);
-
     // Query for the recipe using the provided recipeId
     const recipeData = await collec.findOne({ title: recipeName });
     // Close the MongoDB connection
-    client.close();
```

Comment:
@Mpho-vincent-makgwe this is the other place, please update.

Reply:
Follow the pattern
try {
} catch {
} finally {
client.close()
}

Reply:
Used finally to close the client